flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -1070,80 +1039,75 @@ struct SymbolDependenceDepth {
       std::vector<std::vector<lower::pft::Variable>> &vars)
       : vars{vars} {}
 
-  const semantics::Symbol *setHasGlobalParticipant(
-      const std::vector<semantics::EquivalenceObject> &set,
-      const llvm::DenseSet<const semantics::Symbol *> &globals) {
-    for (const auto &eqv : set)
-      if (globals.find(&eqv.symbol) != globals.end())
-        return &eqv.symbol;
-    return nullptr;
-  }
-
-  std::pair<IntervalSet::iterator, std::size_t>
-  getIntervalForSet(IntervalSet &intervals,
-                    const std::vector<semantics::EquivalenceObject> &set,
-                    const llvm::DenseSet<const semantics::Symbol *> &globals) {
-    for (const auto &eqv : set)
-      if (globals.find(&eqv.symbol) == globals.end()) {
-        auto off = eqv.symbol.offset();
-        return {intervals.find(off), off};
-      }
-    return {intervals.end(), 0};
-  }
-
   // Analyze the equivalence sets. This analysis need not be performed when the
   // scope has no equivalence sets.
   void analyzeAliases(const semantics::Scope &scope) {
-    IntervalSet::Allocator allocator;
-    IntervalSet intervals(allocator);
-    llvm::DenseSet<const semantics::Symbol *> globals;
-
-    // 1. Collect the offset ranges which have aliasing.
-    for (const auto &set : scope.equivalenceSets())
-      for (const auto &eqv : set) {
-        const auto &sym = eqv.symbol;
-        if (symbolIsGlobal(sym)) {
-          // This symbol's offset is into a COMMON block rather than the stack
-          // frame proxy, so do not add it to the interval map. Reprocess the
-          // sets in step 2.
-          globals.insert(&sym);
-          continue;
-        }
-        aliasSyms.insert(&sym);
-        intervals.merge(sym.offset(), sym.offset() + sym.size() - 1);
+    Fortran::lower::IntervalSet intervals;
+    llvm::DenseMap<std::size_t, llvm::SmallVector<const semantics::Symbol *, 8>>
+        aliasSets;
```

Comment:
A map type variable named `aliasSets` might be confusing, how about `aliasGroups` or something else?

Reply:
I think of these as mathematical sets. Variables that are required to alias by Fortran semantics are placed in a set relationship with one another. The programmer can specify some number _n_ of these sets, so we have to track them.